src/ProtocolGateway.Core/Mqtt/MqttAdapter.cs

```diff
@@ -1011,6 +1011,11 @@ async Task CloseServiceConnection(IChannelHandlerContext context, Exception caus
             if (!this.ConnectedToService)
             {
                 // closure happened before IoT Hub connection was established or it was initiated due to disconnect
+                if (cause != null)
+                {
+                    CommonEventSource.Log.Error("Connection was unexpectedly closed or not established.", cause, this.ChannelId, this.Id);
```

Comment:
The message is misleading. I'd say smth like $"Connection closed while not fully connected. Scope: {causeScope}". We may end up here in three distinct cases:

- Connect failed before we connected upstream
- Because of error in upstream connection, it has been torn down already
- Remote peer closed connection

`cause` may contain additional info about the error at this point:
https://github.com/Azure/azure-iot-protocol-gateway/blob/25742f893fde5fdf54265711de3d464480e306e8/src/ProtocolGateway.Core/Mqtt/MqttAdapter.cs#L955-L957
In particular, if `cause.Data` contains `OperationScopeExceptionDataKey` we should include that in the log (it may not be present but error will be different in this case).